src/iuse.cpp

```diff
@@ -8293,3 +8293,35 @@ int iuse::saw_barrel( player *p, item *, bool, const tripoint& )
 
     return 0;
 }
+
+int iuse::washclothes( player *p, item *it, bool, const tripoint& )
+{
+    if( it->charges < it->type->charges_to_use() ) {
+        p->add_msg_if_player( _( "You need a soap to use this." ) );
+        return 0;
+    }
+    
+    const inventory &crafting_inv = p->crafting_inventory();
+    if( !crafting_inv.has_amount( "water", 40 ) && !crafting_inv.has_amount( "water_clean", 40 ) ) {
```

Comment:
I presume this is a requirement for 40 charges of water or clean water? If so, then even standing on the river bank still gives the message about water need.